### PR TITLE
Intermittent test docs

### LINKED ISSUE #1182: Create strategy for handling intermittent tests on the CI
## 🧹 Tech Debt

- Document workflows
- Add needed infrastructure

Potential solution:
- Tests that show intermittency must be handled
  - Either fixed
  - Moved to intermittent CI with logging and a linked issue
  - Temporarily disabled with a linked issue

### PR DESCRIPTION
**Pull request by rajsite**
# Pull Request

## 🤨 Rationale

Document the nimble strategy for handling intermittent tests.
Fixes #1182

## 👩‍💻 Implementation

Added generic docs to the top-level Contributing document.

## 🧪 Testing

N/A

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -160,3 +160,23 @@ Example: Add a monorepo package `nimble-tokens` as a dependency to another monor
 ```bash
 npm install @ni/nimble-tokens --workspace=@ni/nimble-components
 ```
+
+## Handling intermittent test failures
```
Comment (rajsite) [inline CONTRIBUTING.md:164]: I'll bring it up in stand-up to get high-level feedback from the team

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -160,3 +160,23 @@ Example: Add a monorepo package `nimble-tokens` as a dependency to another monor
...
+
+Intermittent test failures can be a huge drain on productivity as they can cause unrelated failures in builds and block merging PRs or creating releases. Particularly when accepting contributions from different timezones and from contributors outside of the design team without the tribal knowledge of what tests fail intermittently.
```
Comment (jattasNI) [inline CONTRIBUTING.md:166]: nit: "design team" reads like interaction/visual design to me
```suggestion
Intermittent test failures can be a huge drain on productivity as they can cause unrelated failures in builds and block merging PRs or creating releases. Particularly when accepting contributions from different timezones and from contributors outside of the Nimble team without the tribal knowledge of what tests fail intermittently.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -160,3 +160,23 @@ Example: Add a monorepo package `nimble-tokens` as a dependency to another monor
...
+
+The general policy is that if intermittent failures are found they should be addressed in main immediately.
+
+Some resolutions for an intermittent test are:
+
+1. Immediately submit a PR to address the underlying issue of the failure. This can be done if the change can be approved and merged by end of day of when the issue was discovered and if there is high confidence in the change.
+2. If the underlying issue is not well-known or the fix does not give high confidence in resolving the intermittent test then the test should be disabled and a tech debt issue created to handle the intermittent test. Then the new issue itself must be handled as follows.
```
Comment (jattasNI) [inline CONTRIBUTING.md:173]:
```suggestion
2. If the underlying issue is not well-known or the fix does not give high confidence in resolving the intermittent test then the test should be disabled and a tech debt issue created to handle the intermittent test. The disabled test should include a comment linking to the corresponding issue. Then the new issue itself must be handled as follows.
```

**Review by jattasNI — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (rajsite) [reply CONTRIBUTING.md:164]: Got team buy-in at standup 👍

**Pull request merged**